rpmio/rpmkeyring.c

```diff
@@ -229,6 +229,28 @@ char * rpmPubkeyBase64(rpmPubkey key)
     return enc;
 }
 
+rpmRC rpmPubkeyMerge(rpmPubkey oldkey, rpmPubkey newkey, rpmPubkey *mergedkeyp)
+{
+    rpmPubkey mergedkey = NULL;
+    uint8_t *mergedpkt = NULL;
+    size_t mergedpktlen = 0;
+    rpmRC rc;
+
+    pthread_rwlock_rdlock(&oldkey->lock);
+    pthread_rwlock_rdlock(&newkey->lock);
```

Comment:
I don't know. I'm not even sure that we need to lock them but found it more safe. Panu?

Reply:
I don't think we have a defined locking order for anything in rpm. The locks on they keys are probably a bit hokey to begin with because librpm and threads is playing with fire anyway.

Reply:
A simple locking order would be to lock according to the memory address.  But, that is probably out of scope for this PR.